mlir/lib/ExecutionEngine/CRunnerUtils.cpp

```diff
@@ -176,6 +177,16 @@ extern "C" void rtdrand(void *g) {
   delete generator;
 }
 
+extern "C" void _mlir_ciface_shuffle(StridedMemRefType<uint64_t, 1> *m,
+                                     void *g) {
+  std::mt19937 *generator = static_cast<std::mt19937 *>(g);
+  uint64_t s = m->sizes[0];
+  std::vector<uint64_t> arr(s);
+  std::iota(arr.begin(), arr.end(), 0);
+  std::shuffle(arr.begin(), arr.end(), *generator);
```

Comment:
can't we copy directly into the array's buffer? Since shuffle seems to work in place, you can to the itoa in the right place right away